src/quo/components/profile/collectible_list_item/view.cljs

```diff
@@ -54,56 +64,83 @@
-       :image       avatar-image-src}]
-     [rn/view {:style {:width 8}}]]
-    [text/text
-     {:size            :paragraph-1
-      :weight          :semi-bold
-      :ellipsize-mode  :tail
-      :number-of-lines 1
-      :style           style/card-detail-text}
-     collectible-name]]])
+           collectible-name]]
+         :else [rn/view
+                {:style (style/avatar-container (:avatar-loaded? state))}
+                [:<>
+                 [collection-avatar/view
+                  {:size        :size-20
+                   :on-load-end #(set-state (fn [prev-state] (assoc prev-state :avatar-loaded? true)))
+                   :image       avatar-image-src}]
+                 [rn/view {:style {:width 8}}]]
+                [text/text
+                 {:size            :paragraph-1
+                  :weight          :semi-bold
+                  :ellipsize-mode  :tail
+                  :number-of-lines 1
+                  :style           style/card-detail-text}
+                 collectible-name]])])
 
 (defn- card-view
   [{:keys [avatar-image-src collectible-name community? counter state set-state
            gradient-color-index image-src supported-file?]}]
-  (let [theme (quo.theme/use-theme)]
+  (let [theme (quo.theme/use-theme)
+        loading-opacity  (reanimated/use-shared-value (if supported-file? 1 0))
+        loaded-opacity  (reanimated/use-shared-value (if supported-file? 0 1))
+        [load-time set-load-time] (rn/use-state (datetime/now))]
+
     [rn/view {:style (style/card-view-container theme)}
      [rn/view {:style {:aspect-ratio 1}}
       (cond
         (:image-error? state)
         [fallback-view
-         {:theme theme
+         {:loaded-opacity loaded-opacity
+          :theme theme
           :label (i18n/label :t/cant-fetch-info)}]
 
         (not supported-file?)
         [fallback-view
-         {:theme theme
+         {:loaded-opacity loaded-opacity
+          :theme theme
           :label (i18n/label :t/unsupported-file)}]
 
         (not (:image-loaded? state))
         [loading-image
-         {:theme                theme
+         {:loading-opacity loading-opacity
+          :theme                theme
           :gradient-color-index gradient-color-index}])
       (when supported-file?
-        [rn/view {:style {:aspect-ratio 1}}
+        [reanimated/view {:style  (reanimated/apply-animations-to-style
+                                   {:opacity loaded-opacity}
+                                   {:aspect-ratio 1})}
          [rn/image
-          {:style       style/image
-           :on-load-end #(set-state (fn [prev-state] (assoc prev-state :image-loaded? true)))
-           :on-error    #(set-state (fn [prev-state] (assoc prev-state :image-error? true)))
+          {:style      style/image
+           :on-load-start #(set-load-time (fn [start-time] (- (datetime/now) start-time)))
+
+
+           :on-load-end (if (> load-time 200)
```

Comment:
`on-load-start` and `on-load-end` can be moved to external functions. It has readability benefits and the potential to save renders.

Reply:
Yep for sure, I will add those types of details once I'm happy with the approach. This was just first pass 👌